consensus/src/consensus/request_response.rs

```diff
@@ -22,45 +27,109 @@ impl<N: Network> Consensus<N> {
         blockchain: &Arc<RwLock<Blockchain>>,
     ) {
         let stream = network.receive_requests::<RequestMacroChain>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_MACRO_CHAIN,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestBatchSet>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_BATCH_SET,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestHistoryChunk>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_HISTORY_CHUNK,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestBlock>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_BLOCK,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestMissingBlocks>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_MISSING_BLOCKS,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestHead>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_HEAD,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
     }
 
-    pub(crate) fn request_handler<Req: Handle<Req::Response> + Request>(
+    pub(crate) fn request_handler<Req: Handle<Req::Response, N> + Request>(
         network: &Arc<N>,
         stream: BoxStream<'static, (Req, N::RequestId, N::PeerId)>,
         blockchain: &Arc<RwLock<Blockchain>>,
+        max_requests: u16,
+        block_range: u32,
     ) -> impl Future<Output = ()> {
         let blockchain = Arc::clone(blockchain);
         let network = Arc::clone(network);
+        let peers_request_limit = Arc::new(futures::lock::Mutex::new(HashMap::with_capacity(
+            network.get_peers().len() * 5,
```

Comment:
This capacity was set in an arbitrary manner, I also considered just leaving it to whatever is the default of the structure. Suggestions?

Reply:
It did not show me this comment while reviewing.

The default size should be fine. `network.get_peers().len()` should also be fine, as peers will likely send requests at some point while connected and there will not be many (if any) when this executes. 

Also note that these get initialized very early, so I would probably just go with default size. Another option might be the peer count floor a node would try to maintain, as eventually that number of peers should be reached at least.